api/tests/Api/CampCollaborations/CreateCampCollaborationTest.php

```diff
@@ -10,17 +10,96 @@
  * @internal
  */
 class CreateCampCollaborationTest extends ECampApiTestCase {
-    // TODO security tests when not logged in or not collaborator
     // TODO input filter tests
     // TODO validation tests
+    // TODO create a camp collaboration for someone else
 
-    public function testCreateCampCollaborationIsAllowedForCollaborator() {
+    public function testCreateCampCollaborationIsDeniedForAnonymousUser() {
+        static::createBasicClient()->request('POST', '/camp_collaborations', ['json' => $this->getExampleWritePayload([
+            'user' => $this->getIriFor('user4unrelated'),
+        ])]);
+
+        $this->assertResponseStatusCodeSame(401);
+        $this->assertJsonContains([
+            'code' => 401,
+            'message' => 'JWT Token not found',
+        ]);
+    }
+
+    public function testCreateCampCollaborationIsNotPossibleForUnrelatedUserBecauseCampIsNotReadable() {
+        static::createClientWithCredentials(['username' => static::$fixtures['user4unrelated']->username])
+            ->request('POST', '/camp_collaborations', ['json' => $this->getExampleWritePayload([
+                'user' => $this->getIriFor('user4unrelated'),
+            ])])
+        ;
+        $this->assertResponseStatusCodeSame(400);
```

Comment:
Wieso nicht 404?

Reply:
Gute Frage. Die `ItemNotFoundException` aus dem `IriConverter` wird [hier](https://github.com/api-platform/core/blob/2.6/src/Serializer/AbstractItemNormalizer.php#L471) in eine `UnexpectedValueException` umgebogen, und ich verstehe nicht ganz warum das nötig sein sollte, auch nicht aus dem Kontext als das eingeführt wurde. Könnte man allenfalls bei API Platform als Verbesserungsvorschlag anbringen. Persönlich ist mir aber hauptsächlich wichtig, dass eine 4xx Response kommt.

Reply:
Ist mir irgendwo auch aufgefallen, dass nicht alle exceptions richtig gemappt werden.
Man kann zutäzliches mapping im services.yaml konfigurieren.
Ich würde sagen, schauen wir uns später mal an, wenn es uns langweilig ist und das das wichtigste wird.